From akpm@linux-foundation.org Thu Apr 26 00:58:14 2007
From: Akinobu Mita <akinobu.mita@gmail.com>
Date: Thu, 26 Apr 2007 00:12:09 -0700
Subject: mod_sysfs_setup() doesn't return errno when kobject_add_dir() failure occurs
To: greg@kroah.com
Cc: akpm@linux-foundation.org, akinobu.mita@gmail.com
Message-ID: <200704260712.l3Q7C9FG023452@shell0.pdx.osdl.net>


From: Akinobu Mita <akinobu.mita@gmail.com>

mod_sysfs_setup() doesn't return an errno when kobject_add_dir() for module
"holders" directory fails.  So caller of mod_sysfs_setup() will keep going
and get oops.

Signed-off-by: Akinobu Mita <akinobu.mita@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 kernel/module.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/kernel/module.c
+++ b/kernel/module.c
@@ -1148,8 +1148,10 @@ int mod_sysfs_setup(struct module *mod,
 		goto out;
 
 	mod->holders_dir = kobject_add_dir(&mod->mkobj.kobj, "holders");
-	if (!mod->holders_dir)
+	if (!mod->holders_dir) {
+		err = -ENOMEM;
 		goto out_unreg;
+	}
 
 	err = module_param_sysfs_setup(mod, kparam, num_params);
 	if (err)
